Strata/DDM/AST.lean

```diff
@@ -244,6 +244,25 @@ def asCommaSepList : Arg → Array Arg
 
 end Arg
 
+/--
+Source location information in the DDM is defined
+by a range of bytes in a UTF-8 string with the input
+Line/column information can be construced from a
+`Lean.FileMap`
+-/
+structure SourceLoc where
```

Comment:
I wonder if it'd be clearer to call this `SourceRange` or `SourceSpan`, since it's not a single location.